google/cloud/bigtable/client_options.cc

```diff
@@ -93,9 +93,9 @@ ClientOptions& ClientOptions::set_connection_pool_size(std::size_t size) {
 }
 
 std::string ClientOptions::UserAgentPrefix() {
-  std::string agent = "gcloud-cpp/" + version_string() + " " +
-                      google::cloud::internal::compiler();
-  return agent;
+  static auto const kUserAgentPrefix =
+      google::cloud::internal::UserAgentPrefix();
```

Comment:
These ones seem to break the "no static-duration objects unless they are trivially destructible" convention.

Reply:
The common way to avoid the issue that @devbww mentions is to heap-allocate the object so that the static thing is only a pointer or reference. Like:

```cc
  static auto const* kUserAgentPrefix = new auto(google::cloud::internal::UserAgentPrefix());
  return *kUserAgentPrefix;
```

Reply:
I decided to move the static optimization into the internal function.... although now I am noticing that I used `static auto* const` instead of `static auto const*` as @devjgm suggested...

Reply:
I think we want both consts because the pointer and the thing it points to are const.

Reply:
Done?